controller/util/cf_v2_client_test.go

```diff
@@ -14,22 +14,44 @@ limitations under the License.
 package util
 
 import (
+	"net/http"
+	"net/http/httptest"
 	"testing"
 
 	model "github.com/kubernetes-incubator/service-catalog/model/service_broker"
 	scmodel "github.com/kubernetes-incubator/service-catalog/model/service_controller"
 )
 
-func TestUpdateServiceInstance(t *testing.T) {
-	cli := CreateCFV2BrokerClient(&scmodel.ServiceBroker{})
+func TestCreateServiceInstance(t *testing.T) {
+	fakeBroker := newFakeBroker()
+	defer fakeBroker.Close()
+	const id = "testID"
+	req := &model.ServiceInstanceRequest{
+		OrgID:             "testOrgID",
+		PlanID:            "testPlanID",
+		ServiceID:         "testSvcID",
+		SpaceID:           "testSpaceID",
+		Parameters:        map[string]interface{}{},
+		AcceptsIncomplete: false,
+	}
+
+	cl := CreateCFV2BrokerClient(&scmodel.ServiceBroker{
+		BrokerURL: fakeBroker.URLStr(),
+	})
+
+	_, err := cl.CreateServiceInstance(id, req)
```

Comment:
will we want to validate that the URL path was correct, body, etc.?
and perhaps provide a response body ahead of time?

how fancy-programmable should the fake broker be (I am not sure, just wondering).
There is a fine line between having an actual fake broker, or writing up small pieces of the fake broker functionality that handle just the right slice of URL paths to satisfy the test.

Generic fake broker will need to be quite generic to handle all necessary test cases, which may include maintaining state across calls.